src/keydra/providers/salesforce_marketing_cloud.py

```diff
@@ -0,0 +1,145 @@
+from typing import NamedTuple
+import boto3
+import boto3.session
+
+from keydra.clients.salesforce_marketing_cloud import SalesforceMarketingCloudClient
+
+from keydra.clients.aws.secretsmanager import SecretsManagerClient
+
+from keydra.providers.base import BaseProvider
+from keydra.providers.base import exponential_backoff_retry
+
+from keydra.exceptions import ConfigException, DistributionException
+from keydra.exceptions import RotationException
+
+from keydra.logging import get_logger
+
+LOGGER = get_logger()
+
+PASS_LENGTH = 32
+
+
```

Comment:
Keen to get some thoughts on this. It's feasible to combine this with the existing Salesforce provider. I opted against it to keep it more maintainable if Salesforce Sales Cloud differs greately from Salesforce Marketing Cloud in the long run.

Reply:
I agree tbh. I think for now, separate makes more sense. I get the felling that SFMC is somewhat separate from core SF in terms of platform, so feel the providers should be separate too. I assume that you did not find you had to duplicate large chunks of code? 

We've been here before with the Splunk and Splunk HEC providers - we opted for separate there as well after some debate.

Reply:
👍  Duplicated the structure, but did change a few things given the Client for SFMC is vastly different.